lib/pod/command/trunk/push.rb

```diff
@@ -140,17 +140,28 @@ def validate_podspec
         end
 
         def update_master_repo
-          sources_manager = if defined?(Pod::SourcesManager)
-                              Pod::SourcesManager
-                            else
-                              config.sources_manager
-                            end
           if sources_manager.master_repo_functional?
-            sources_manager.update('master')
+            sources_manager.update(master_repo_name)
           else
             Setup.invoke
           end
         end
+
+        def master_repo_name
+          sources_manager.master.first.name
```

Comment:
where is the `master` method or attribute defined? I cant seem to find it

Reply:
https://github.com/CocoaPods/Core/blob/d0cab9c32f575e4d8a747a0136f59a0c676e7522/lib/cocoapods-core/source/manager.rb#L70

Reply:
Dunno why this is an array, but that's a question for another day 😭